test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,13 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if (isMutatorCommand(inputCommand)) {
```

Comment:
Good attempt at improving code readability!
Perhaps a another solution would be to make use of the `isMutating()` method that you have defined. 
```java
if(new Parser().parseCommand(inputCommand).isMutating()) {
    assertEquals(addressBook, saveFile.load())
}
```

Suppose you have new mutating commands and that you forgot to update `isMutatorCommand(...)` the test would still pass without having tested the new mutating commands.